vyper/venom/venom_to_assembly.py

```diff
@@ -543,7 +559,7 @@ def _generate_evm_for_instruction(
 
         # Step 6: Emit instructions output operands (if any)
         if inst.output is not None:
-            if "call" in inst.opcode and inst.output not in next_liveness:
+            if cleanup_needed and opcode != "param" and inst.output not in next_liveness:
```

Comment:
i guess `"param"` instructions are handled already in clean_unused_params? can we just remove that function now?